lib/loader.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * (C) Copyright 2016 o2r project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+var c = require('../config/config');
+const debug = require('debug')('loader');
+const Docker = require('dockerode');
+const steps = require('../lib/loader-steps');
+const randomstring = require('randomstring');
+
+/**
+ * Create Loader to handle given request and response
```

Comment:
add a comment on the callback function (it is either passed an error or the id of the successfully loaded compendium)